File: solutions/kotlin/raindrops/1/src/main/kotlin/Raindrops.kt
object Raindrops {
    fun convert(number: Int) : String{
        var output = ""
        (1..number)
                .asSequence()
                .filter { number % it == 0 }
                .forEach {
                    when(it){
                        3 -> output += "Pling"
                        5 -> output += "Plang"
                        7 -> output += "Plong"
                    }
                }
        if (output.isEmpty()){
            output = number.toString()
        }
        
        return output
    }
}

File: solutions/python/raindrops/1/raindrops.py
def convert(number):
    raindrop_speak = ""
    mapping = {
        3: "Pling",
        5: "Plang",
        7: "Plong"
    }
    
    # get factors of number
    for x in range(1, number + 1):
        if number % x == 0:
            word = mapping.get(x, None)

            if word:
                raindrop_speak += word
    
    return raindrop_speak if raindrop_speak else str(number)